src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs

```diff
@@ -250,172 +180,183 @@ private void DeleteSelectedButton_Click(object sender, RoutedEventArgs e)
+                _graphicsOverlay.Graphics.Add(new Graphic(geometry, GetSymbol(geometry.GeometryType)));
             }
 
-            // Allow the creation of new graphics.
-            EnableGeometryButtons();
-            ToolComboBox.IsEnabled = true;
+            ResetFromEditingSession();
         }
 
         // Stops the geometry editor without saving the geometry stored within.
         private void DiscardButton_Click(object sender, RoutedEventArgs e)
         {
-            if (_selectedGraphic != null)
-            {
-                // Make visisble the pre-existing graphic when editing.
-                _selectedGraphic.IsVisible = true;
-                _selectedGraphic.IsSelected = false;
-                _selectedGraphic = null;
-
-                _geometryEditor.ClearSelection();
-            }
-
             _geometryEditor.Stop();
-
-            // Allow for the creation of new graphics.
-            EnableGeometryButtons();
-            ToolComboBox.IsEnabled = true;
+            ResetFromEditingSession();
         }
 
         // Removes all graphics from the graphics overlay.
         private void DeleteAllButton_Click(object sender, RoutedEventArgs e)
         {
             _graphicsOverlay.Graphics.Clear();
+        }
 
-            // Disable the delete all button as there are no graphics to delete.
-            DeleteAllButton.IsEnabled = false;
-            _geometryCreated = false;
+        private async void MyMapView_GeoViewTapped(object sender, GeoViewInputEventArgs e)
+        {
+            // Return immediately when in an editing session.
+            if (_geometryEditor.IsStarted) return;
 
-            // Allow for the creation of new graphics.
-            EnableGeometryButtons();
-            ToolComboBox.IsEnabled = true;
+            try
+            {
+                // Identify graphics in the graphics overlay using the mouse point.
+                IReadOnlyList<IdentifyGraphicsOverlayResult> results = await MyMapView.IdentifyGraphicsOverlaysAsync(e.Position, 5, false);
+
+                // Try to get the first graphic from the first result.
+                _selectedGraphic = results.FirstOrDefault()?.Graphics?.FirstOrDefault();
+            }
+            catch (Exception ex)
+            {
+                // Report exceptions.
+                MessageBox.Show("Editing error: " + ex.Message, ex.GetType().Name, MessageBoxButton.OK, MessageBoxImage.Error);
+
+                ResetFromEditingSession();
+                return;
+            }
+
+            // Return since no graphic was selected.
+            if (_selectedGraphic == null) return;
+
+            _selectedGraphic.IsSelected = true;
+
+            // Configure the UI depending on the geometry type.
+            GeometryType geometryType = _selectedGraphic.Geometry.GeometryType;
+            if (geometryType == GeometryType.Point || geometryType == GeometryType.Multipoint)
+            {
+                ToolComboBox.SelectedIndex = 0;
```

Comment:
I think we want VertexTool to always be the default.
Allowing users to change it to Freehand when Polyline/Polygon seem enough. Which you're doing by handling the `ComboBox.SelectionChanged` event

Reply:
If user has freehand tool enabled, selects a polyline/polygon, no need to change the tool to vertex tool. But if they have freehand tool enabled, select a point/multipoint, it is necessary to change the tool to a vertex tool.